compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -76,8 +76,10 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   }
 
   /** A function type or closure with `implicit`, `erased`, or `given` modifiers */
-  class FunctionWithMods(args: List[Tree], body: Tree, val mods: Modifiers)(implicit @constructorOnly src: SourceFile)
-    extends Function(args, body)
+  class FunctionWithMods(args: List[Tree], body: Tree, val mods: Modifiers, val isErased: List[Boolean])(implicit @constructorOnly src: SourceFile)
```

Comment:
The name `isErased` clashes with the concept of an erased definition (such as inline definition, erased parameter) and type erasure. I would change the name `isErased` to `erasedParam`.